### PR TITLE
[jit] Add support for unlimited size/number of arguments to the dyn call code

### PR DESCRIPTION
**Pull request by vargaz**

### CODE REVIEW

File: mono/mini/mini-amd64.h [hunk 1]
```diff
@@ -270,15 +270,15 @@ typedef struct {
...
 
-#define DYN_CALL_STACK_ARGS 6
-
 typedef struct {
-	mgreg_t regs [PARAM_REGS + DYN_CALL_STACK_ARGS];
 	mgreg_t res;
 	guint8 *ret;
 	double fregs [8];
 	mgreg_t has_fp;
+	mgreg_t nstack_args;
```
Comment (lewurm) [inline mono/mini/mini-amd64.h]: `mono/metadata/object-offsets.h` must be updated

**Comment by lewurm** [comment]
does the test case work on the other arches (x86, arm)?

**Comment by vargaz** [comment]
Not yet.

**Comment by lewurm** [comment]
build interpreter

**Comment by vargaz** [comment]
build

**Pull request merged**